lib/sycamore/sycamore/transforms/text_extraction/ocr_models.py

```diff
@@ -0,0 +1,216 @@
+        for obj in ocr_output:
+            if obj["bbox"] and not obj["bbox"].is_empty() and obj["text"] and len(obj["text"]) > 0:
+                text = Element()
+                text.type = "text"
+                text.bbox = BoundingBox(
+                    obj["bbox"].x1 / width,
+                    obj["bbox"].y1 / height,
+                    obj["bbox"].x2 / width,
+                    obj["bbox"].y2 / height,
+                )
+                text.text_representation = obj["text"]
+                texts.append(text)
+        return texts
+
+    @timetrace("OCR Document Extraction")
+    def extract_document(
+        self, filename: Union[str, IOBase], hash_key: str, use_cache=False, **kwargs
+    ) -> List[List[Element]]:
+        cached_result = ocr_cache.get(hash_key) if use_cache else None
+        if cached_result:
+            logger.info(f"Cache Hit for OCR. Cache hit-rate is {ocr_cache.get_hit_rate()}")
+            return cached_result
+        else:
+            with tempfile.TemporaryDirectory() as tempdirname:  # type: ignore
+                filename = cast(str, filename)
+                images = kwargs.get("images")
+                generator = (image for image in images) if images else pdf_to_image_files(filename, Path(tempdirname))
+                pages = []
+                for path in generator:
+                    if isinstance(path, Image.Image):
+                        image = path
+                    else:
+                        image = Image.open(path).convert("RGB")
+                    ocr_output = self.get_boxes_and_text(image)
+                    width, height = image.size
+                    texts: List[Element] = []
+                    for obj in ocr_output:
+                        if obj["bbox"] and not obj["bbox"].is_empty() and obj["text"] and len(obj["text"]) > 0:
+                            text = Element()
+                            text.type = "text"
+                            text.bbox = BoundingBox(
+                                obj["bbox"].x1 / width,
+                                obj["bbox"].y1 / height,
+                                obj["bbox"].x2 / width,
+                                obj["bbox"].y2 / height,
+                            )
+                            text.text_representation = obj["text"]
+                            texts.append(text)
+
+                    pages.append(texts)
+                if use_cache:
+                    logger.info("Cache Miss for OCR. Storing the result to the cache.")
+                    ocr_cache.set(hash_key, pages)
+                return pages
+
+
+class EasyOcr(OcrModel):
+    @requires_modules("easyocr", extra="local-inference")
+    def __init__(self, lang_list=["en"]):
+        import easyocr
+
+        self.reader = easyocr.Reader(lang_list=lang_list)
+
+    def get_text(self, image: Image.Image) -> str:
+        image_bytes = BytesIO()
+        image.save(image_bytes, format="PNG")
```

Comment:
Do we need PNG?  Compression is slow and we have enough space for these transient files.